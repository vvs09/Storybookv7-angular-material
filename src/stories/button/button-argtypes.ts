export const buttonArgTypes= {
    color: {
      control: 'radio',
      options: ['accent', 'primary', 'warn'],
      description: 'Theme color palette for the component.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'null' },
        type: {
          summary: 'string',
        },
      },
    },
    disabled: {
      control: 'boolean',
      description:
        'Whether the checkbox is disabled. This fully overrides the implementation provided by mixinDisabled, but the mixin is still required because mixinTabIndex requires it.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'false' },
        type: {
          summary: 'boolean',
        },
      },
    },
    disableRipple: {
      control: 'boolean',
      description: 'Whether ripples are disabled.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'false' },
        type: {
          summary: 'boolean',
        },
      },
    },
    buttonType: {
      control: 'select',
      options: [
        'mat-button',
        'mat-raised-button',
        'mat-stroked-button',
        'mat-flat-button',
      ],
      description: 'Type of buton.',
      table: {
        defaultValue: { summary: 'mat-button' },
        type: {
          summary: 'string',
        },
      },
    },
    label: {
      control: 'text',
      description: 'Button label.',
      table: {
        defaultValue: { summary: 'button' },
        type: {
          summary: 'string',
        },
      },
    },
    ariaDisabled: {
      control: 'boolean',
      description: 'aria-disabled value of the button.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'undefined' },
        type: {
          summary: 'boolean | undefined',
        },
      },
    },
    disabledInteractive: {
      control: 'boolean',
      description:
        'Natively disabled buttons prevent focus and any pointer events from reaching the button. In some scenarios this might not be desirable, because it can prevent users from finding out why the button is disabled (e.g. via tooltip). Enabling this input will change the button so that it is styled to be disabled and will be marked as aria-disabled, but it will allow the button to receive events and focus. Note that by enabling this, you need to set the tabindex yourself if the button isnt meant to be tabbable and you have to prevent the button action (e.g. form submissions).',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'false' },
        type: {
          summary: 'boolean',
        },
      },
    },
  }