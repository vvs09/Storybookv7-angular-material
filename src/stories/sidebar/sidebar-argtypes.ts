export const sideBarArgTypes = {
  autoFocus: {
    control: 'boolean',
    description:
      'Whether the drawer should focus the first focusable element automatically when opened. Defaults to false in when mode is set to side, otherwise defaults to true. If explicitly enabled, focus will be moved into the sidenav in side mode as well.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: 'false' },
      type: {
        summary: 'AutoFocusTarget | string | boolean',
      },
    },
  },
  disableClose: {
    control: 'boolean',
    description:
      'Whether the drawer can be closed with the escape key or by clicking on the backdrop.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: 'false' },
      type: {
        summary: 'boolean',
      },
    },
  },
  fixedBottomGap: {
    control: 'number',
    description:
      'The gap between the bottom of the sidenav and the bottom of the viewport when the sidenav is in fixed mode.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'number',
      },
    },
  },
  fixedInViewport: {
    control: 'boolean',
    description: 'Whether the sidenav is fixed in the viewport.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: 'false' },
      type: {
        summary: 'boolean',
      },
    },
  },
  position: {
    control: 'radio',
    options: ['start', 'end'],
    description: 'The side that the drawer is attached to.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: "'start' | 'end'",
      },
    },
  },
  fixedTopGap: {
    control: 'number',
    description:
      'The gap between the top of the sidenav and the top of the viewport when the sidenav is in fixed mode.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'number',
      },
    },
  },
  mode: {
    control: 'radio',
    options: ['over', 'push', 'side'],
    description: "Mode of the drawer; one of 'over', 'push' or 'side'.",
    table: {
      category: 'INPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: "'over' |'push' | 'side'",
      },
    },
  },
  opened: {
    control: 'boolean',
    description:
      'Whether the drawer is opened. We overload this because we trigger an event when it starts or end.',
    table: {
      category: 'INPUTS',
      defaultValue: { summary: 'false' },
      type: {
        summary: 'boolean',
      },
    },
  },
  closedStart: {
    control: '-',
    description: 'Event emitted when the drawer has started closing.',
    table: {
      category: 'OUTPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'Observable<void>',
      },
    },
  },
  onPositionChanged: {
    control: '-',
    description: "Event emitted when the drawer's position changes.",
    table: {
      category: 'OUTPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'EventEmitter<void>',
      },
    },
  },
  openedChange: {
    control: '-',
    description: 'Event emitted when the drawer open state is changed.',
    table: {
      category: 'OUTPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: ' EventEmitter<boolean>',
      },
    },
  },
  openedStart: {
    control: '-',
    description: 'Event emitted when the drawer has started opening.',
    table: {
      category: 'OUTPUTS',
      defaultValue: { summary: '-' },
      type: {
        summary: 'Observable<void>',
      },
    },
  },
};
