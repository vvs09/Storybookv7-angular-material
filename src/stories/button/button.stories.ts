import type { Meta, StoryObj } from '@storybook/angular';
import { argsToTemplate, moduleMetadata } from '@storybook/angular';
import { MatButtonModule, MatButton } from '@angular/material/button';
import { buttonArgTypes } from './button-argtypes';

const meta: Meta<MatButton & { label: string; buttonType: string }> = {
  title: 'Components/Button',
  component: MatButton,
  tags: ['autodocs'],
  argTypes: {
    ...buttonArgTypes,
  },
  decorators: [
    moduleMetadata({
      imports: [MatButtonModule],
    }),
  ],
  render: (args) => ({
    props: args,
    template: `
    <button  *ngIf="buttonType === 'mat-button'" mat-button  ${argsToTemplate(
      args
    )} >{{label}}</button>
    <button *ngIf="buttonType === 'mat-raised-button' " mat-raised-button ${argsToTemplate(
      args
    )} >{{label}}</button>
    <button *ngIf="buttonType === 'mat-stroked-button' " mat-stroked-button ${argsToTemplate(
      args
    )} >{{label}}</button>
    <button *ngIf="buttonType === 'mat-flat-button' "  mat-flat-button ${argsToTemplate(
      args
    )} >{{label}}</button>
\    `,
  }),
};

export default meta;
type Story = StoryObj<MatButton & { label: string; buttonType: string }>;

export const accent: Story = {
  args: {
    color: 'accent',
    disabled: false,
    ariaDisabled: false,
    disabledInteractive: false,
    disableRipple: false,
    label: 'Button',
    buttonType: 'mat-button',
  },
};

export const Primary: Story = {
  args: {
    ...accent.args,
    color: 'primary',
  },
};

export const Warn: Story = {
  args: {
    ...accent.args,
    color: 'warn',
  },
};

export const Disabled: Story = {
  args: {
    ...accent.args,
    disabled: true,
  },
};

export const matRaisedButtonAccent: Story = {
  args: {
    ...accent.args,
    buttonType: 'mat-raised-button',
  },
};

export const matRaisedButtonPrimary: Story = {
  args: {
    ...matRaisedButtonAccent.args,
    color: 'primary',
  },
};

export const matRaisedButtonWarn: Story = {
  args: {
    ...matRaisedButtonAccent.args,
    color: 'warn',
  },
};

export const MatRaisedButtonDisabled: Story = {
  args: {
    ...matRaisedButtonAccent.args,
    disabled: true,
  },
};

export const MatStrokedButtonAccent: Story = {
  args: {
    ...accent.args,
    buttonType: 'mat-stroked-button',
  },
};

export const MatStrokedButtonPrimary: Story = {
  args: {
    ...MatStrokedButtonAccent.args,
    color: 'primary',
  },
};

export const MatStrokedButtonWarn: Story = {
  args: {
    ...MatStrokedButtonAccent.args,
    color: 'warn',
  },
};

export const MatStrokedButtonDisabled: Story = {
  args: {
    ...MatStrokedButtonAccent.args,
    disabled: true,
  },
};

export const MatFlatButtonAccent: Story = {
  args: {
    ...accent.args,
    buttonType: 'mat-flat-button',
  },
};

export const MatFlatButtonPrimary: Story = {
  args: {
    ...MatFlatButtonAccent.args,
    color: 'primary',
  },
};

export const MatFlatButtonWarn: Story = {
  args: {
    ...MatFlatButtonAccent.args,
    color: 'warn',
  },
};

export const MatFlatButtonDisabled: Story = {
  args: {
    ...MatFlatButtonAccent.args,
    disabled: true,
  },
};
