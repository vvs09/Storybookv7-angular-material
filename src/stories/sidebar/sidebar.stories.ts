import type { Meta, StoryObj } from '@storybook/angular';
import { argsToTemplate, moduleMetadata } from '@storybook/angular';
import { MatSidenavModule, MatSidenav } from '@angular/material/sidenav';
import { MatButtonModule } from '@angular/material/button';
import { sideBarArgTypes } from './sidebar-argtypes';

const meta = {
  title: 'Components/Sidenav',
  component: MatSidenav,
  tags: ['autodocs'],
  argTypes: {
    ...sideBarArgTypes,
  },
  decorators: [
    moduleMetadata({
      imports: [MatSidenavModule, MatButtonModule],
    }),
  ],
  render: (args:any) => ({
    props: args,
    styles: [
      `
     .sidebar{
      width: 500px;
      height: 300px;
      border: 1px solid rgba(0, 0, 0, 0.5);"
     }
    .sidebar-button{
     display: flex;
     height: 100%;
     align-items: center;
     justify-content: center;
     }
     .
     `,
    ],
    template: `
    <mat-drawer-container class="sidebar" autosize>
  <mat-drawer #drawer style=" padding: 20px;" ${argsToTemplate(args)}  >
    <p>Auto-resizing sidenav</p>
    @if (showFiller) {
      <p>Lorem, ipsum dolor sit amet consectetur.</p>
    }
    <button (click)="showFiller = !showFiller" mat-raised-button>
      Toggle extra text
    </button>
  </mat-drawer>
  <div class="sidebar-button">
    <button type="button" mat-button (click)="drawer.toggle()">
      Toggle sidenav
    </button>
  </div>
</mat-drawer-container>

   `,
  }),
};

export default meta;
type Story = StoryObj<MatSidenav>;

export const Basic: Story = {
  args: {
    mode: 'over',
    opened: false,
    autoFocus: false,
    disableClose: false,
    fixedInViewport: false,
    position: 'start',
  },
};

export const ModePush: Story = {
  args: {
    ...Basic.args,
    mode: 'push',
  },
};

export const Warn: Story = {
  args: {
    ...Basic.args,
    mode: 'side',
  },
};

export const PositionEnd: Story = {
  args: {
    ...Basic.args,
    position: 'end',
  },
};

export const AutoFocus: Story = {
  args: {
    ...Basic.args,
    autoFocus: true,
  },
};

export const DisableClose: Story = {
  args: {
    ...Basic.args,
    disableClose: true,
  },
};

export const Open: Story = {
  args: {
    ...Basic.args,
    opened: true,
  },
};

export const FixedInViewport: Story = {
  args: {
    ...Basic.args,
    fixedInViewport: true,
  },
};
