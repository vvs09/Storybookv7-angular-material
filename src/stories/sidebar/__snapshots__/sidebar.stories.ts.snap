// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Sidenav AutoFocus smoke-test 1`] = `
<storybook-root _nghost-ng-c650243922
                ng-version="17.2.2"
>
  <mat-drawer-container _ngcontent-ng-c650243922
                        autosize
                        class="mat-drawer-container sidebar"
                        ng-reflect-autosize
  >
    <div class="mat-drawer-backdrop ng-star-inserted">
    </div>
    <div tabindex="0"
         class="cdk-visually-hidden cdk-focus-trap-anchor"
         aria-hidden="true"
    >
    </div>
    <mat-drawer _ngcontent-ng-c650243922
                tabindex="-1"
                class="mat-drawer ng-tns-c3519248442-0 ng-trigger ng-trigger-transform mat-drawer-over ng-star-inserted"
                style="padding: 20px; box-shadow: none; visibility: hidden;"
                ng-reflect-mode="over"
                ng-reflect-opened="false"
                ng-reflect-auto-focus="true"
                ng-reflect-disable-close="false"
                ng-reflect-position="start"
    >
      <div cdkscrollable
           class="mat-drawer-inner-container ng-tns-c3519248442-0"
      >
        <p _ngcontent-ng-c650243922
           class="ng-tns-c3519248442-0"
        >
          Auto-resizing sidenav
        </p>
        <button _ngcontent-ng-c650243922
                mat-raised-button
                mat-ripple-loader-uninitialized
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                class="mdc-button mdc-button--raised mat-mdc-raised-button ng-tns-c3519248442-0 mat-unthemed mat-mdc-button-base"
        >
          <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
          </span>
          <span class="mdc-button__label">
            Toggle extra text
          </span>
          <span class="mat-mdc-focus-indicator">
          </span>
          <span class="mat-mdc-button-touch-target">
          </span>
        </button>
      </div>
    </mat-drawer>
    <div tabindex="0"
         class="cdk-visually-hidden cdk-focus-trap-anchor"
         aria-hidden="true"
    >
    </div>
    <mat-drawer-content class="mat-drawer-content ng-star-inserted">
      <div _ngcontent-ng-c650243922
           class="sidebar-button"
      >
        <button _ngcontent-ng-c650243922
                type="button"
                mat-button
                mat-ripple-loader-uninitialized
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                class="mdc-button mat-mdc-button mat-unthemed mat-mdc-button-base"
        >
          <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
          </span>
          <span class="mdc-button__label">
            Toggle sidenav
          </span>
          <span class="mat-mdc-focus-indicator">
          </span>
          <span class="mat-mdc-button-touch-target">
          </span>
        </button>
      </div>
    </mat-drawer-content>
  </mat-drawer-container>
</storybook-root>
`;

exports[`Components/Sidenav Basic smoke-test 1`] = `
<storybook-root _nghost-ng-c650243922
                ng-version="17.2.2"
>
  <mat-drawer-container _ngcontent-ng-c650243922
                        autosize
                        class="mat-drawer-container sidebar"
                        ng-reflect-autosize
  >
    <div class="mat-drawer-backdrop ng-star-inserted">
    </div>
    <div tabindex="0"
         class="cdk-visually-hidden cdk-focus-trap-anchor"
         aria-hidden="true"
    >
    </div>
    <mat-drawer _ngcontent-ng-c650243922
                tabindex="-1"
                class="mat-drawer ng-tns-c3519248442-0 ng-trigger ng-trigger-transform mat-drawer-over ng-star-inserted"
                style="padding: 20px; box-shadow: none; visibility: hidden;"
                ng-reflect-mode="over"
                ng-reflect-opened="false"
                ng-reflect-auto-focus="false"
                ng-reflect-disable-close="false"
                ng-reflect-position="start"
    >
      <div cdkscrollable
           class="mat-drawer-inner-container ng-tns-c3519248442-0"
      >
        <p _ngcontent-ng-c650243922
           class="ng-tns-c3519248442-0"
        >
          Auto-resizing sidenav
        </p>
        <button _ngcontent-ng-c650243922
                mat-raised-button
                mat-ripple-loader-uninitialized
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                class="mdc-button mdc-button--raised mat-mdc-raised-button ng-tns-c3519248442-0 mat-unthemed mat-mdc-button-base"
        >
          <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
          </span>
          <span class="mdc-button__label">
            Toggle extra text
          </span>
          <span class="mat-mdc-focus-indicator">
          </span>
          <span class="mat-mdc-button-touch-target">
          </span>
        </button>
      </div>
    </mat-drawer>
    <div tabindex="0"
         class="cdk-visually-hidden cdk-focus-trap-anchor"
         aria-hidden="true"
    >
    </div>
    <mat-drawer-content class="mat-drawer-content ng-star-inserted">
      <div _ngcontent-ng-c650243922
           class="sidebar-button"
      >
        <button _ngcontent-ng-c650243922
                type="button"
                mat-button
                mat-ripple-loader-uninitialized
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                class="mdc-button mat-mdc-button mat-unthemed mat-mdc-button-base"
        >
          <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
          </span>
          <span class="mdc-button__label">
            Toggle sidenav
          </span>
          <span class="mat-mdc-focus-indicator">
          </span>
          <span class="mat-mdc-button-touch-target">
          </span>
        </button>
      </div>
    </mat-drawer-content>
  </mat-drawer-container>
</storybook-root>
`;

exports[`Components/Sidenav DisableClose smoke-test 1`] = `
<storybook-root _nghost-ng-c650243922
                ng-version="17.2.2"
>
  <mat-drawer-container _ngcontent-ng-c650243922
                        autosize
                        class="mat-drawer-container sidebar"
                        ng-reflect-autosize
  >
    <div class="mat-drawer-backdrop ng-star-inserted">
    </div>
    <div tabindex="0"
         class="cdk-visually-hidden cdk-focus-trap-anchor"
         aria-hidden="true"
    >
    </div>
    <mat-drawer _ngcontent-ng-c650243922
                tabindex="-1"
                class="mat-drawer ng-tns-c3519248442-0 ng-trigger ng-trigger-transform mat-drawer-over ng-star-inserted"
                style="padding: 20px; box-shadow: none; visibility: hidden;"
                ng-reflect-mode="over"
                ng-reflect-opened="false"
                ng-reflect-auto-focus="false"
                ng-reflect-disable-close="true"
                ng-reflect-position="start"
    >
      <div cdkscrollable
           class="mat-drawer-inner-container ng-tns-c3519248442-0"
      >
        <p _ngcontent-ng-c650243922
           class="ng-tns-c3519248442-0"
        >
          Auto-resizing sidenav
        </p>
        <button _ngcontent-ng-c650243922
                mat-raised-button
                mat-ripple-loader-uninitialized
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                class="mdc-button mdc-button--raised mat-mdc-raised-button ng-tns-c3519248442-0 mat-unthemed mat-mdc-button-base"
        >
          <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
          </span>
          <span class="mdc-button__label">
            Toggle extra text
          </span>
          <span class="mat-mdc-focus-indicator">
          </span>
          <span class="mat-mdc-button-touch-target">
          </span>
        </button>
      </div>
    </mat-drawer>
    <div tabindex="0"
         class="cdk-visually-hidden cdk-focus-trap-anchor"
         aria-hidden="true"
    >
    </div>
    <mat-drawer-content class="mat-drawer-content ng-star-inserted">
      <div _ngcontent-ng-c650243922
           class="sidebar-button"
      >
        <button _ngcontent-ng-c650243922
                type="button"
                mat-button
                mat-ripple-loader-uninitialized
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                class="mdc-button mat-mdc-button mat-unthemed mat-mdc-button-base"
        >
          <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
          </span>
          <span class="mdc-button__label">
            Toggle sidenav
          </span>
          <span class="mat-mdc-focus-indicator">
          </span>
          <span class="mat-mdc-button-touch-target">
          </span>
        </button>
      </div>
    </mat-drawer-content>
  </mat-drawer-container>
</storybook-root>
`;

exports[`Components/Sidenav FixedInViewport smoke-test 1`] = `
<storybook-root _nghost-ng-c650243922
                ng-version="17.2.2"
>
  <mat-drawer-container _ngcontent-ng-c650243922
                        autosize
                        class="mat-drawer-container sidebar"
                        ng-reflect-autosize
  >
    <div class="mat-drawer-backdrop ng-star-inserted">
    </div>
    <div tabindex="0"
         class="cdk-visually-hidden cdk-focus-trap-anchor"
         aria-hidden="true"
    >
    </div>
    <mat-drawer _ngcontent-ng-c650243922
                tabindex="-1"
                class="mat-drawer ng-tns-c3519248442-0 ng-trigger ng-trigger-transform mat-drawer-over ng-star-inserted"
                style="padding: 20px; box-shadow: none; visibility: hidden;"
                ng-reflect-mode="over"
                ng-reflect-opened="false"
                ng-reflect-auto-focus="false"
                ng-reflect-disable-close="false"
                ng-reflect-position="start"
    >
      <div cdkscrollable
           class="mat-drawer-inner-container ng-tns-c3519248442-0"
      >
        <p _ngcontent-ng-c650243922
           class="ng-tns-c3519248442-0"
        >
          Auto-resizing sidenav
        </p>
        <button _ngcontent-ng-c650243922
                mat-raised-button
                mat-ripple-loader-uninitialized
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                class="mdc-button mdc-button--raised mat-mdc-raised-button ng-tns-c3519248442-0 mat-unthemed mat-mdc-button-base"
        >
          <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
          </span>
          <span class="mdc-button__label">
            Toggle extra text
          </span>
          <span class="mat-mdc-focus-indicator">
          </span>
          <span class="mat-mdc-button-touch-target">
          </span>
        </button>
      </div>
    </mat-drawer>
    <div tabindex="0"
         class="cdk-visually-hidden cdk-focus-trap-anchor"
         aria-hidden="true"
    >
    </div>
    <mat-drawer-content class="mat-drawer-content ng-star-inserted">
      <div _ngcontent-ng-c650243922
           class="sidebar-button"
      >
        <button _ngcontent-ng-c650243922
                type="button"
                mat-button
                mat-ripple-loader-uninitialized
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                class="mdc-button mat-mdc-button mat-unthemed mat-mdc-button-base"
        >
          <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
          </span>
          <span class="mdc-button__label">
            Toggle sidenav
          </span>
          <span class="mat-mdc-focus-indicator">
          </span>
          <span class="mat-mdc-button-touch-target">
          </span>
        </button>
      </div>
    </mat-drawer-content>
  </mat-drawer-container>
</storybook-root>
`;

exports[`Components/Sidenav ModePush smoke-test 1`] = `
<storybook-root _nghost-ng-c650243922
                ng-version="17.2.2"
>
  <mat-drawer-container _ngcontent-ng-c650243922
                        autosize
                        class="mat-drawer-container sidebar"
                        ng-reflect-autosize
  >
    <div class="mat-drawer-backdrop ng-star-inserted">
    </div>
    <div tabindex="0"
         class="cdk-visually-hidden cdk-focus-trap-anchor"
         aria-hidden="true"
    >
    </div>
    <mat-drawer _ngcontent-ng-c650243922
                tabindex="-1"
                class="mat-drawer ng-tns-c3519248442-0 ng-trigger ng-trigger-transform mat-drawer-push ng-star-inserted"
                style="padding: 20px; box-shadow: none; visibility: hidden;"
                ng-reflect-mode="push"
                ng-reflect-opened="false"
                ng-reflect-auto-focus="false"
                ng-reflect-disable-close="false"
                ng-reflect-position="start"
    >
      <div cdkscrollable
           class="mat-drawer-inner-container ng-tns-c3519248442-0"
      >
        <p _ngcontent-ng-c650243922
           class="ng-tns-c3519248442-0"
        >
          Auto-resizing sidenav
        </p>
        <button _ngcontent-ng-c650243922
                mat-raised-button
                mat-ripple-loader-uninitialized
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                class="mdc-button mdc-button--raised mat-mdc-raised-button ng-tns-c3519248442-0 mat-unthemed mat-mdc-button-base"
        >
          <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
          </span>
          <span class="mdc-button__label">
            Toggle extra text
          </span>
          <span class="mat-mdc-focus-indicator">
          </span>
          <span class="mat-mdc-button-touch-target">
          </span>
        </button>
      </div>
    </mat-drawer>
    <div tabindex="0"
         class="cdk-visually-hidden cdk-focus-trap-anchor"
         aria-hidden="true"
    >
    </div>
    <mat-drawer-content class="mat-drawer-content ng-star-inserted">
      <div _ngcontent-ng-c650243922
           class="sidebar-button"
      >
        <button _ngcontent-ng-c650243922
                type="button"
                mat-button
                mat-ripple-loader-uninitialized
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                class="mdc-button mat-mdc-button mat-unthemed mat-mdc-button-base"
        >
          <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
          </span>
          <span class="mdc-button__label">
            Toggle sidenav
          </span>
          <span class="mat-mdc-focus-indicator">
          </span>
          <span class="mat-mdc-button-touch-target">
          </span>
        </button>
      </div>
    </mat-drawer-content>
  </mat-drawer-container>
</storybook-root>
`;

exports[`Components/Sidenav Open smoke-test 1`] = `
<storybook-root _nghost-ng-c650243922
                ng-version="17.2.2"
>
  <mat-drawer-container _ngcontent-ng-c650243922
                        autosize
                        class="mat-drawer-container sidebar mat-drawer-container-has-open"
                        ng-reflect-autosize
  >
    <div class="mat-drawer-backdrop mat-drawer-shown ng-star-inserted">
    </div>
    <div tabindex="0"
         class="cdk-visually-hidden cdk-focus-trap-anchor"
         aria-hidden="true"
    >
    </div>
    <mat-drawer _ngcontent-ng-c650243922
                tabindex="-1"
                class="mat-drawer ng-tns-c3519248442-0 ng-trigger ng-trigger-transform mat-drawer-over mat-drawer-opened ng-star-inserted"
                style="padding: 20px; transform: none; visibility: visible;"
                ng-reflect-mode="over"
                ng-reflect-opened="true"
                ng-reflect-auto-focus="false"
                ng-reflect-disable-close="false"
                ng-reflect-position="start"
    >
      <div cdkscrollable
           class="mat-drawer-inner-container ng-tns-c3519248442-0"
      >
        <p _ngcontent-ng-c650243922
           class="ng-tns-c3519248442-0"
        >
          Auto-resizing sidenav
        </p>
        <button _ngcontent-ng-c650243922
                mat-raised-button
                mat-ripple-loader-uninitialized
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                class="mdc-button mdc-button--raised mat-mdc-raised-button ng-tns-c3519248442-0 mat-unthemed mat-mdc-button-base"
        >
          <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
          </span>
          <span class="mdc-button__label">
            Toggle extra text
          </span>
          <span class="mat-mdc-focus-indicator">
          </span>
          <span class="mat-mdc-button-touch-target">
          </span>
        </button>
      </div>
    </mat-drawer>
    <div tabindex="0"
         class="cdk-visually-hidden cdk-focus-trap-anchor"
         aria-hidden="true"
    >
    </div>
    <mat-drawer-content class="mat-drawer-content ng-star-inserted">
      <div _ngcontent-ng-c650243922
           class="sidebar-button"
      >
        <button _ngcontent-ng-c650243922
                type="button"
                mat-button
                mat-ripple-loader-uninitialized
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                class="mdc-button mat-mdc-button mat-unthemed mat-mdc-button-base"
        >
          <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
          </span>
          <span class="mdc-button__label">
            Toggle sidenav
          </span>
          <span class="mat-mdc-focus-indicator">
          </span>
          <span class="mat-mdc-button-touch-target">
          </span>
        </button>
      </div>
    </mat-drawer-content>
  </mat-drawer-container>
</storybook-root>
`;

exports[`Components/Sidenav PositionEnd smoke-test 1`] = `
<storybook-root _nghost-ng-c650243922
                ng-version="17.2.2"
>
  <mat-drawer-container _ngcontent-ng-c650243922
                        autosize
                        class="mat-drawer-container sidebar"
                        ng-reflect-autosize
  >
    <div class="mat-drawer-backdrop ng-star-inserted">
    </div>
    <mat-drawer-content class="mat-drawer-content ng-star-inserted">
      <div _ngcontent-ng-c650243922
           class="sidebar-button"
      >
        <button _ngcontent-ng-c650243922
                type="button"
                mat-button
                mat-ripple-loader-uninitialized
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                class="mdc-button mat-mdc-button mat-unthemed mat-mdc-button-base"
        >
          <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
          </span>
          <span class="mdc-button__label">
            Toggle sidenav
          </span>
          <span class="mat-mdc-focus-indicator">
          </span>
          <span class="mat-mdc-button-touch-target">
          </span>
        </button>
      </div>
    </mat-drawer-content>
    <div tabindex="0"
         class="cdk-visually-hidden cdk-focus-trap-anchor"
         aria-hidden="true"
    >
    </div>
    <mat-drawer _ngcontent-ng-c650243922
                tabindex="-1"
                class="mat-drawer ng-tns-c3519248442-0 ng-trigger ng-trigger-transform mat-drawer-end mat-drawer-over ng-star-inserted"
                style="padding: 20px; box-shadow: none; visibility: hidden;"
                ng-reflect-mode="over"
                ng-reflect-opened="false"
                ng-reflect-auto-focus="false"
                ng-reflect-disable-close="false"
                ng-reflect-position="end"
    >
      <div cdkscrollable
           class="mat-drawer-inner-container ng-tns-c3519248442-0"
      >
        <p _ngcontent-ng-c650243922
           class="ng-tns-c3519248442-0"
        >
          Auto-resizing sidenav
        </p>
        <button _ngcontent-ng-c650243922
                mat-raised-button
                mat-ripple-loader-uninitialized
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                class="mdc-button mdc-button--raised mat-mdc-raised-button ng-tns-c3519248442-0 mat-unthemed mat-mdc-button-base"
        >
          <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
          </span>
          <span class="mdc-button__label">
            Toggle extra text
          </span>
          <span class="mat-mdc-focus-indicator">
          </span>
          <span class="mat-mdc-button-touch-target">
          </span>
        </button>
      </div>
    </mat-drawer>
    <div tabindex="0"
         class="cdk-visually-hidden cdk-focus-trap-anchor"
         aria-hidden="true"
    >
    </div>
  </mat-drawer-container>
</storybook-root>
`;

exports[`Components/Sidenav Warn smoke-test 1`] = `
<storybook-root _nghost-ng-c650243922
                ng-version="17.2.2"
>
  <mat-drawer-container _ngcontent-ng-c650243922
                        autosize
                        class="mat-drawer-container sidebar"
                        ng-reflect-autosize
  >
    <div class="cdk-visually-hidden cdk-focus-trap-anchor"
         aria-hidden="true"
    >
    </div>
    <mat-drawer _ngcontent-ng-c650243922
                tabindex="-1"
                class="mat-drawer ng-tns-c3519248442-0 ng-trigger ng-trigger-transform mat-drawer-side ng-star-inserted"
                style="padding: 20px; box-shadow: none; visibility: hidden;"
                ng-reflect-mode="side"
                ng-reflect-opened="false"
                ng-reflect-auto-focus="false"
                ng-reflect-disable-close="false"
                ng-reflect-position="start"
    >
      <div cdkscrollable
           class="mat-drawer-inner-container ng-tns-c3519248442-0"
      >
        <p _ngcontent-ng-c650243922
           class="ng-tns-c3519248442-0"
        >
          Auto-resizing sidenav
        </p>
        <button _ngcontent-ng-c650243922
                mat-raised-button
                mat-ripple-loader-uninitialized
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                class="mdc-button mdc-button--raised mat-mdc-raised-button ng-tns-c3519248442-0 mat-unthemed mat-mdc-button-base"
        >
          <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
          </span>
          <span class="mdc-button__label">
            Toggle extra text
          </span>
          <span class="mat-mdc-focus-indicator">
          </span>
          <span class="mat-mdc-button-touch-target">
          </span>
        </button>
      </div>
    </mat-drawer>
    <div class="cdk-visually-hidden cdk-focus-trap-anchor"
         aria-hidden="true"
    >
    </div>
    <mat-drawer-content class="mat-drawer-content ng-star-inserted">
      <div _ngcontent-ng-c650243922
           class="sidebar-button"
      >
        <button _ngcontent-ng-c650243922
                type="button"
                mat-button
                mat-ripple-loader-uninitialized
                mat-ripple-loader-class-name="mat-mdc-button-ripple"
                class="mdc-button mat-mdc-button mat-unthemed mat-mdc-button-base"
        >
          <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
          </span>
          <span class="mdc-button__label">
            Toggle sidenav
          </span>
          <span class="mat-mdc-focus-indicator">
          </span>
          <span class="mat-mdc-button-touch-target">
          </span>
        </button>
      </div>
    </mat-drawer-content>
  </mat-drawer-container>
</storybook-root>
`;
